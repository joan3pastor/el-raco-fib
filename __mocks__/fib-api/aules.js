export const quadri_actual = {
    "id": "2020Q1",
    "url": "https://api.fib.upc.edu/v2/quadrimestres/2020Q1/",
    "actual": "S",
    "actual_horaris": "S",
    "classes": "https://api.fib.upc.edu/v2/quadrimestres/2020Q1/classes/",
    "examens": "https://api.fib.upc.edu/v2/quadrimestres/2020Q1/examens/",
    "assignatures": "https://api.fib.upc.edu/v2/quadrimestres/2020Q1/assignatures/"
};

export const aules = {
    "count": 51,
    "next": null,
    "previous": null,
    "results": [
        {
            "id": "A1S101"
        },
        {
            "id": "A3204"
        },
        {
            "id": "A4002"
        },
        {
            "id": "A5001"
        },
        {
            "id": "A5002"
        },
        {
            "id": "A5101"
        },
        {
            "id": "A5102"
        },
        {
            "id": "A5103"
        },
        {
            "id": "A5104"
        },
        {
            "id": "A5105"
        },
        {
            "id": "A5106"
        },
        {
            "id": "A5201"
        },
        {
            "id": "A5202"
        },
        {
            "id": "A5203"
        },
        {
            "id": "A5204"
        },
        {
            "id": "A5205"
        },
        {
            "id": "A5206"
        },
        {
            "id": "A5E01"
        },
        {
            "id": "A5E02"
        },
        {
            "id": "A5S108"
        },
        {
            "id": "A5S109"
        },
        {
            "id": "A5S112"
        },
        {
            "id": "A5S113"
        },
        {
            "id": "A6001"
        },
        {
            "id": "A6002"
        },
        {
            "id": "A6101"
        },
        {
            "id": "A6102"
        },
        {
            "id": "A6103"
        },
        {
            "id": "A6104"
        },
        {
            "id": "A6105"
        },
        {
            "id": "A6106"
        },
        {
            "id": "A6201"
        },
        {
            "id": "A6202"
        },
        {
            "id": "A6203"
        },
        {
            "id": "A6204"
        },
        {
            "id": "A6205"
        },
        {
            "id": "A6206"
        },
        {
            "id": "A6E01"
        },
        {
            "id": "A6E02"
        },
        {
            "id": "B5S201"
        },
        {
            "id": "B5S202"
        },
        {
            "id": "C5S202"
        },
        {
            "id": "C5S203"
        },
        {
            "id": "C6S301"
        },
        {
            "id": "C6S302"
        },
        {
            "id": "C6S303"
        },
        {
            "id": "C6S306"
        },
        {
            "id": "C6S308"
        },
        {
            "id": "C6S309"
        },
        {
            "id": "D6003"
        },
        {
            "id": "Online"
        }
    ]
};

