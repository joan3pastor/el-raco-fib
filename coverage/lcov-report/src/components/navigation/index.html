<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/navigation</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\..\index.html">All files</a> src/components/navigation
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">78.01% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>337/432</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">86.43% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>191/221</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">54.61% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>77/141</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">75.74% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>281/371</span>
      </div>
    </div>
    <p class="quiet">
      Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
    </p>
  </div>
  <div class='status-line medium'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="Assignatures.js"><a href="Assignatures.js.html">Assignatures.js</a></td>
	<td data-value="82.35" class="pic high"><div class="chart"><div class="cover-fill" style="width: 82%;"></div><div class="cover-empty" style="width:18%;"></div></div></td>
	<td data-value="82.35" class="pct high">82.35%</td>
	<td data-value="34" class="abs high">28/34</td>
	<td data-value="90.91" class="pct high">90.91%</td>
	<td data-value="22" class="abs high">20/22</td>
	<td data-value="75" class="pct medium">75%</td>
	<td data-value="16" class="abs medium">12/16</td>
	<td data-value="82.35" class="pct high">82.35%</td>
	<td data-value="34" class="abs high">28/34</td>
	</tr>

<tr>
	<td class="file high" data-value="Avisos.js"><a href="Avisos.js.html">Avisos.js</a></td>
	<td data-value="81.98" class="pic high"><div class="chart"><div class="cover-fill" style="width: 81%;"></div><div class="cover-empty" style="width:19%;"></div></div></td>
	<td data-value="81.98" class="pct high">81.98%</td>
	<td data-value="172" class="abs high">141/172</td>
	<td data-value="91.45" class="pct high">91.45%</td>
	<td data-value="117" class="abs high">107/117</td>
	<td data-value="54.29" class="pct medium">54.29%</td>
	<td data-value="35" class="abs medium">19/35</td>
	<td data-value="77.17" class="pct medium">77.17%</td>
	<td data-value="127" class="abs medium">98/127</td>
	</tr>

<tr>
	<td class="file low" data-value="Drawer.js"><a href="Drawer.js.html">Drawer.js</a></td>
	<td data-value="47.37" class="pic low"><div class="chart"><div class="cover-fill" style="width: 47%;"></div><div class="cover-empty" style="width:53%;"></div></div></td>
	<td data-value="47.37" class="pct low">47.37%</td>
	<td data-value="19" class="abs low">9/19</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="8" class="abs high">8/8</td>
	<td data-value="30.77" class="pct low">30.77%</td>
	<td data-value="13" class="abs low">4/13</td>
	<td data-value="47.37" class="pct low">47.37%</td>
	<td data-value="19" class="abs low">9/19</td>
	</tr>

<tr>
	<td class="file high" data-value="Events.js"><a href="Events.js.html">Events.js</a></td>
	<td data-value="81.03" class="pic high"><div class="chart"><div class="cover-fill" style="width: 81%;"></div><div class="cover-empty" style="width:19%;"></div></div></td>
	<td data-value="81.03" class="pct high">81.03%</td>
	<td data-value="58" class="abs high">47/58</td>
	<td data-value="67.65" class="pct medium">67.65%</td>
	<td data-value="34" class="abs medium">23/34</td>
	<td data-value="86.67" class="pct high">86.67%</td>
	<td data-value="15" class="abs high">13/15</td>
	<td data-value="78.43" class="pct medium">78.43%</td>
	<td data-value="51" class="abs medium">40/51</td>
	</tr>

<tr>
	<td class="file low" data-value="ExternalLinks.js"><a href="ExternalLinks.js.html">ExternalLinks.js</a></td>
	<td data-value="20" class="pic low"><div class="chart"><div class="cover-fill" style="width: 20%;"></div><div class="cover-empty" style="width:80%;"></div></div></td>
	<td data-value="20" class="pct low">20%</td>
	<td data-value="15" class="abs low">3/15</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="14.29" class="pct low">14.29%</td>
	<td data-value="14" class="abs low">2/14</td>
	<td data-value="20" class="pct low">20%</td>
	<td data-value="15" class="abs low">3/15</td>
	</tr>

<tr>
	<td class="file low" data-value="Feedback.js"><a href="Feedback.js.html">Feedback.js</a></td>
	<td data-value="40" class="pic low"><div class="chart"><div class="cover-fill" style="width: 40%;"></div><div class="cover-empty" style="width:60%;"></div></div></td>
	<td data-value="40" class="pct low">40%</td>
	<td data-value="5" class="abs low">2/5</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="25" class="pct low">25%</td>
	<td data-value="4" class="abs low">1/4</td>
	<td data-value="40" class="pct low">40%</td>
	<td data-value="5" class="abs low">2/5</td>
	</tr>

<tr>
	<td class="file high" data-value="Horari.js"><a href="Horari.js.html">Horari.js</a></td>
	<td data-value="89.86" class="pic high"><div class="chart"><div class="cover-fill" style="width: 89%;"></div><div class="cover-empty" style="width:11%;"></div></div></td>
	<td data-value="89.86" class="pct high">89.86%</td>
	<td data-value="69" class="abs high">62/69</td>
	<td data-value="86.11" class="pct high">86.11%</td>
	<td data-value="36" class="abs high">31/36</td>
	<td data-value="69.23" class="pct medium">69.23%</td>
	<td data-value="13" class="abs medium">9/13</td>
	<td data-value="90.91" class="pct high">90.91%</td>
	<td data-value="66" class="abs high">60/66</td>
	</tr>

<tr>
	<td class="file high" data-value="Labs.js"><a href="Labs.js.html">Labs.js</a></td>
	<td data-value="81.25" class="pic high"><div class="chart"><div class="cover-fill" style="width: 81%;"></div><div class="cover-empty" style="width:19%;"></div></div></td>
	<td data-value="81.25" class="pct high">81.25%</td>
	<td data-value="16" class="abs high">13/16</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="2" class="abs medium">1/2</td>
	<td data-value="66.67" class="pct medium">66.67%</td>
	<td data-value="9" class="abs medium">6/9</td>
	<td data-value="85.71" class="pct high">85.71%</td>
	<td data-value="14" class="abs high">12/14</td>
	</tr>

<tr>
	<td class="file medium" data-value="Noticies.js"><a href="Noticies.js.html">Noticies.js</a></td>
	<td data-value="71.43" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 71%;"></div><div class="cover-empty" style="width:29%;"></div></div></td>
	<td data-value="71.43" class="pct medium">71.43%</td>
	<td data-value="42" class="abs medium">30/42</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="2" class="abs medium">1/2</td>
	<td data-value="45" class="pct low">45%</td>
	<td data-value="20" class="abs low">9/20</td>
	<td data-value="71.05" class="pct medium">71.05%</td>
	<td data-value="38" class="abs medium">27/38</td>
	</tr>

<tr>
	<td class="file high" data-value="SalesBibl.js"><a href="SalesBibl.js.html">SalesBibl.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri Jul 17 2020 06:12:00 GMT+0200 (GMT+02:00)
</div>
</div>
<script src="..\..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\..\sorter.js"></script>
<script src="..\..\..\block-navigation.js"></script>
</body>
</html>
