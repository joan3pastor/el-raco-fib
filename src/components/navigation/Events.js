import React, { Component } from 'react';
import { Text, View, ScrollView } from 'react-native';
import { connect } from 'react-redux';
import _ from 'lodash';

import {  } from '../../actions';

export class Events extends Component {

  filterExamens() {
    var assigs = _.uniqBy(this.props.assig.results, 'id').map((assig) => assig.id);
    var allEvents = [...this.props.examens.results.filter((examen) => {
      return assigs.find((assig) => { return assig == examen.assig; }) 
    }).map((examen) => {
      return {
        tipus: 'examen', 
        assig: examen.assig, 
        dia: /\d+-\d+-\d+/.exec(examen.inici)[0], 
        horaInici: /\d+:\d+/.exec(examen.inici)[0], 
        horaFi: /\d+:\d+/.exec(examen.fi)[0], 
        tipusExamen: examen.tipus};
    })];
    return allEvents;
  }

  filterEvents(events) {
    var eventsFI = [];
    var allEvents = [...events, ...this.props.events.results.filter(({nom}) =>  nom == 'FESTIU' || nom == 'CANVI DIA' || nom == 'VACANCES' || nom == 'FESTA FIB' || nom == 'FI-EXAMENS' || nom == 'CURS').map((event) => {
      if (event.inici != event.fi && event.nom != 'FESTIU') {
        eventsFI.push({
          tipus: 'event',
          nom: 'FI-' + event.nom,
          dia: /\d+-\d+-\d+/.exec(event.fi)[0], 
          hora: '00:00',
        });
        return {
          tipus: 'event',
          nom: 'INICI-' + event.nom,
          dia: /\d+-\d+-\d+/.exec(event.inici)[0], 
          hora: '00:00',
        };

      } else if (event.inici != event.fi && event.nom == 'FESTIU') { //Si és un Festiu de més d'un día
        let dateInici = new Date(/\d+-\d+-\d+/.exec(event.inici)[0]);
        let dateFi = new Date(/\d+-\d+-\d+/.exec(event.fi)[0]);
        while (dateFi.getTime() > dateInici.getTime()) {

          dateInici.setDate(dateInici.getDate()+1);
          let diaAny = String(dateInici.getFullYear());
          let diaMes = String(dateInici.getMonth()+1);
          let diaDia = String(dateInici.getDate());
          if (diaMes.length == 1) diaMes = '0'+diaMes;
          if (diaDia.length == 1) diaDia = '0'+diaDia;
          eventsFI.push({
            tipus: 'event',
            nom: event.nom,
            dia: `${diaAny}-${diaMes}-${diaDia}`, 
            hora: '00:00',
          });
        }
        return {
          tipus: 'event',
          nom: event.nom,
          dia: /\d+-\d+-\d+/.exec(event.inici)[0], 
          hora: '00:00',
        };

      } //Si és un event d'un únic día
      return {
        tipus: 'event',
        nom: event.nom,
        dia: /\d+-\d+-\d+/.exec(event.inici)[0], 
        hora: '00:00',
      };
    })];
    allEvents = [...allEvents, ...eventsFI];

    return allEvents;
  }

  filterByDay(events) {
    var allEvents = _.orderBy(events, ['dia', 'hora'], ['asc', 'asc']).filter((ev) => {
      // * 3 Meses en ms = 7776000000
      // * 1 Dia en ms = 86400000
      let diff = new Date(ev.dia) - new Date();
      if (diff < -86400000) return false;  //Si ya ha pasado 1 día
      if (diff > 7776000000) return false; // Si faltan mas de 3 meses
      return true;
    });
    return allEvents;
  }

  // filterPractiques(events) {
  //   var allEvents = [...events, ...this.props.practiques.results.map((practica) => {
  //     return {
  //       tipus: 'practica',
  //       nom: practica.titol,
  //       assig: practica.codi_asg,
  //       dia: /\d+-\d+-\d+/.exec(practica.data_limit)[0], 
  //       hora: /\d+:\d+/.exec(practica.data_limit)[0],
  //     };
  //   })];
  //   return allEvents;
  // }

  getEvents() {
    //FILTER EXAMENS:
    var allEvents = this.filterExamens();
    //FILTER EVENTS:
    allEvents = this.filterEvents(allEvents);
    //FILTER BY DAY:
    allEvents = this.filterByDay(allEvents);
    return allEvents;
  }

  formatDay(pre) {
    try {
      var ind = pre.split('-');
      return `${ind[2]}/${ind[1]}/${ind[0]}`;
    } catch {
      return "?";
    }
  }

  render() {
    if (_.isUndefined(this.props.examens.results) || _.isUndefined(this.props.events.results) /*|| _.isUndefined(this.props.practiques.results)*/) {
      return (
        <View style={{backgroundColor:'#fafafa', flex:1, alignItems:'center', justifyContent:'center'}}>
          <Text>Carregant propers esdeveniments...</Text>
        </View>
      );
    }

    displayEvents = this.getEvents().map((ev) => {
      if (ev.tipus == 'examen') {
        return (
          <View key={`ex${ev.dia}${ev.inici}${ev.assig}`} style={{flexDirection:'row', backgroundColor:'#fff',  margin:5, marginLeft:15, marginRight:15, borderRadius:5, padding: 12, elevation:1}}>
            <Text style={{color:'#555555'}}>{`${this.formatDay(ev.dia)} ${ev.horaInici}-${ev.horaFi}: `}</Text>
            <Text style={{color:'#555555', fontWeight:'bold'}}>{`${ev.assig} Examen ${ev.tipusExamen=='P' ? 'Parcial' : 'Final'}   `}</Text>
          </View>
        );
      }
      if (ev.tipus == 'event') {
        return (
          <View key={`ev${ev.dia}${ev.nom}`} style={{flexDirection:'row', backgroundColor:'#fff',  margin:5, marginLeft:15, marginRight:15, borderRadius:5, padding: 12, elevation:1}}>
            <Text style={{color:'#555555'}}>{`${this.formatDay(ev.dia)}: `}</Text>
            <Text style={{color:'#555555', fontWeight:'bold'}}>{`${ev.nom}   `}</Text>
          </View>
        );
      }
      if (ev.tipus == 'practica') {
        return (
          <View key={`pr${ev.assig}${ev.nom}`} style={{flexDirection:'row', backgroundColor:'#fff',  margin:5, marginLeft:15, marginRight:15, borderRadius:5, padding: 12, elevation:1}}>
            <Text style={{color:'#555555'}}>{`${this.formatDay(ev.dia)} ${ev.hora}: `}</Text>
            <Text style={{color:'#555555', fontWeight:'bold'}}>{`${ev.assig} Practica ${ev.nom}   `}</Text>
          </View>
        );
      }
      return <View ></View>
    });

    return (
      <ScrollView style={{backgroundColor:'#fafafa', flex:1, paddingTop:10}}>
        {displayEvents}
    <Text style={{margin:10, marginBottom:30}}>{`${this.props.lang.footer}     `}</Text>
      </ScrollView>
    );
  }

}

function mapStateToProps(state) {
  return { 
    examens: state.examens,
    events: state.events,
    practiques: state.practiques,
    assig: state.assig,
    lang: state.lang.events,
  };
}

export default connect(mapStateToProps, {  })(Events);
