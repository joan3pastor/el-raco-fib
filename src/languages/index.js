import {ca} from "./ca.js";
import {es} from "./es.js";
import {en} from "./en.js";

export default {
    ca: ca,
    es: es,
    en: en,
};