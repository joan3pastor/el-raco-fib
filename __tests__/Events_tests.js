import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import React from "react";
import configureStore from "redux-mock-store";
import { shallow } from "enzyme";
//import jest from 'jest';
import axios from "axios";
import sinon from "sinon";
import AxiosMockAdapter from "axios-mock-adapter";

// LANG
import {ca} from "../src/languages/ca";
const lang = ca.events;

// Components
import { Events } from "../src/components/navigation/Events";

// Mocks
import {
  events_buit,
  events,
  event_parsed_old,
  event_parsed_more_3months
} from "../__mocks__/fib-api/events";
import { examens_buit, examens } from "../__mocks__/fib-api/examens";
import {
  assignatures_buit,
  assignatures_5
} from "../__mocks__/fib-api/assignatures";
import {
  default_settings,
  minimalist_settings,
  small_fontsize_settings,
  paleta
} from "../__mocks__/app-settings";

Enzyme.configure({ adapter: new Adapter() });
const middlewares = []; // you can mock any middlewares here if necessary
const mockStore = configureStore(middlewares);

const initialState = {};

beforeEach(() => {});

describe("Testing formatDay method from Events component", () => {
  const wrapper = shallow(
    <Events
      store={mockStore(initialState)}
      events={events_buit}
      assig={assignatures_buit}
      examens={examens_buit}
      lang={lang}
    />,
    { disableLifecycleMethods: true }
  );

  // Method to test
  formatDay = wrapper.instance().formatDay;

  it("formats avis.date into 'DD/MM/AAAA hh:mm", () => {
    expect(
      formatDay.bind({
        props: {lang:lang},
        state: {},
        setState: obj => obj
      })("2020-04-12T21:08:09")
    ).toBe("12T21:08:09/04/2020");
  });

  it("formats avis.date into 'DD/MM/AAAA hh:mm", () => {
    expect(
      formatDay.bind({
        props: {lang:lang},
        state: {},
        setState: obj => obj
      })("2020-06-29T11:48:50")
    ).toBe("29T11:48:50/06/2020");
  });
});

describe("Testing filterExamens method from Events component", () => {
  const wrapper = shallow(
    <Events
      store={mockStore(initialState)}
      events={events_buit}
      assig={assignatures_buit}
      examens={examens_buit}
      lang={lang}
    />,
    { disableLifecycleMethods: false }
  );

  // Method to test
  filterExamens = wrapper.instance().filterExamens;

  it("returns 0 'examens' when there are no subjects", () => {
    context = {
      props: {
        events: events_buit,
        assig: assignatures_buit,
        examens: examens,
        lang:lang,
      },
      state: {},
      setState: obj => obj
    };

    expect(filterExamens.bind(context)()).toMatchInlineSnapshot("Array []");
  });

  it("returns 'examens' from the active subjects", () => {
    context = {
      props: {
        events: events_buit,
        assig: assignatures_5,
        examens: examens,
        lang:lang,
      },
      state: {},
      setState: obj => obj
    };

    expect(filterExamens.bind(context)()).toMatchSnapshot();
  });
});

describe("Testing filterEvents method from Events component", () => {
  const wrapper = shallow(
    <Events
      store={mockStore(initialState)}
      events={events_buit}
      assig={assignatures_buit}
      examens={examens_buit}
      lang={lang}
    />,
    { disableLifecycleMethods: false }
  );

  // Method to test
  filterEvents = wrapper.instance().filterEvents;

  it("returns 0 'events' when <reduxstore>.events is empty", () => {
    context = {
      props: {
        events: events_buit,
        assig: assignatures_buit,
        examens: examens_buit,
        lang:lang,
      },
      state: {},
      setState: obj => obj
    };

    expect(filterEvents.bind(context)([])).toMatchInlineSnapshot("Array []");
  });

  it("returns 'events' from <reduxstore>.events parsed", () => {
    context = {
      props: {
        events: events,
        assig: assignatures_buit,
        examens: examens_buit,
        lang:lang,
      },
      state: {},
      setState: obj => obj
    };

    expect(filterEvents.bind(context)([])).toMatchSnapshot();
  });
});

describe("Testing filterByDay method from Events component", () => {
  const wrapper = shallow(
    <Events
      store={mockStore(initialState)}
      events={events_buit}
      assig={assignatures_buit}
      examens={examens_buit}
      lang={lang}
    />,
    { disableLifecycleMethods: false }
  );

  // Method to test
  filterByDay = wrapper.instance().filterByDay;

  it("returns 0 'events' because the 'event' provided is too old", () => {
    context = {
      props: {},
      state: {},
      setState: obj => obj
    };

    expect(filterByDay.bind(context)(event_parsed_old)).toMatchInlineSnapshot(
      "Array []"
    );
  });

  it("returns 0 'events' because the date from the 'event' provided is in more than 3 months", () => {
    context = {
      props: {},
      state: {},
      setState: obj => obj
    };

    expect(
      filterByDay.bind(context)(event_parsed_more_3months)
    ).toMatchInlineSnapshot("Array []");
  });
});

describe("Testing getEvents method from Events component", () => {
  const wrapper = shallow(
    <Events
      store={mockStore(initialState)}
      events={events_buit}
      assig={assignatures_buit}
      examens={examens_buit}
      lang={lang}
    />,
    { disableLifecycleMethods: false }
  );

  // Method to test
  getEvents = wrapper.instance().getEvents;

  it("gathers all events from filterExamens and filterEvents and filters them by day with filterByDay", () => {
    context = {
      filterExamens: function() {
        return [1, 2];
      },
      filterEvents: function(evs) {
        return [...evs, 3, 4];
      },
      filterByDay: function(evs) {
        return [...evs, 5, 6];
      },
      props: {},
      state: {},
      setState: obj => obj
    };

    expect(getEvents.bind(context)()).toMatchInlineSnapshot(
      `
      Array [
        1,
        2,
        3,
        4,
        5,
        6,
      ]
    `
    );
  });
});

describe("Testing render from Events component", () => {
  it("renders view with 0 events", () => {
    const wrapper = shallow(
      <Events
        store={mockStore(initialState)}
        events={events_buit}
        assig={assignatures_buit}
        examens={examens_buit}
        lang={lang}
      />,
      { disableLifecycleMethods: false }
    );

    expect(wrapper).toMatchSnapshot();
  });

  it("renders view with events and examens", () => {
    const wrapper = shallow(
      <Events
        store={mockStore(initialState)}
        events={events}
        assig={assignatures_5}
        examens={examens}
        lang={lang}
      />,
      { disableLifecycleMethods: false }
    );

    expect(wrapper).toMatchSnapshot();
  });
});
