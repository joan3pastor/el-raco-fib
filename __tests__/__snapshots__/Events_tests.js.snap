// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing filterEvents method from Events component returns 'events' from <reduxstore>.events parsed 1`] = `
Array [
  Object {
    "dia": "2019-09-11",
    "hora": "00:00",
    "nom": "FESTIU",
    "tipus": "event",
  },
  Object {
    "dia": "2019-09-23",
    "hora": "00:00",
    "nom": "FESTIU",
    "tipus": "event",
  },
  Object {
    "dia": "2019-10-31",
    "hora": "00:00",
    "nom": "CANVI DIA",
    "tipus": "event",
  },
  Object {
    "dia": "2019-11-01",
    "hora": "00:00",
    "nom": "FESTIU",
    "tipus": "event",
  },
  Object {
    "dia": "2019-12-06",
    "hora": "00:00",
    "nom": "FESTIU",
    "tipus": "event",
  },
  Object {
    "dia": "2019-12-23",
    "hora": "00:00",
    "nom": "INICI-VACANCES",
    "tipus": "event",
  },
  Object {
    "dia": "2020-02-10",
    "hora": "00:00",
    "nom": "INICI-CURS",
    "tipus": "event",
  },
  Object {
    "dia": "2020-04-06",
    "hora": "00:00",
    "nom": "FESTIU",
    "tipus": "event",
  },
  Object {
    "dia": "2020-04-13",
    "hora": "00:00",
    "nom": "FESTIU",
    "tipus": "event",
  },
  Object {
    "dia": "2020-05-01",
    "hora": "00:00",
    "nom": "FESTIU",
    "tipus": "event",
  },
  Object {
    "dia": "2020-06-01",
    "hora": "00:00",
    "nom": "FESTIU",
    "tipus": "event",
  },
  Object {
    "dia": "2020-06-23",
    "hora": "00:00",
    "nom": "FI-EXAMENS",
    "tipus": "event",
  },
  Object {
    "dia": "2019-09-24",
    "hora": "00:00",
    "nom": "FESTIU",
    "tipus": "event",
  },
  Object {
    "dia": "2020-01-06",
    "hora": "00:00",
    "nom": "FI-VACANCES",
    "tipus": "event",
  },
  Object {
    "dia": "2020-05-29",
    "hora": "00:00",
    "nom": "FI-CURS",
    "tipus": "event",
  },
  Object {
    "dia": "2020-04-07",
    "hora": "00:00",
    "nom": "FESTIU",
    "tipus": "event",
  },
  Object {
    "dia": "2020-04-08",
    "hora": "00:00",
    "nom": "FESTIU",
    "tipus": "event",
  },
  Object {
    "dia": "2020-04-09",
    "hora": "00:00",
    "nom": "FESTIU",
    "tipus": "event",
  },
  Object {
    "dia": "2020-04-10",
    "hora": "00:00",
    "nom": "FESTIU",
    "tipus": "event",
  },
]
`;

exports[`Testing filterExamens method from Events component returns 'examens' from the active subjects 1`] = `
Array [
  Object {
    "assig": "ASW",
    "dia": "2020-06-10",
    "horaFi": "14:30",
    "horaInici": "11:30",
    "tipus": "examen",
    "tipusExamen": "F",
  },
  Object {
    "assig": "ROB",
    "dia": "2020-06-10",
    "horaFi": "20:30",
    "horaInici": "18:30",
    "tipus": "examen",
    "tipusExamen": "F",
  },
]
`;

exports[`Testing render from Events component renders view with 0 events 1`] = `
<ScrollViewMock
  style={
    Object {
      "backgroundColor": "#fafafa",
      "flex": 1,
      "paddingTop": 10,
    }
  }
>
  <Text
    style={
      Object {
        "margin": 10,
        "marginBottom": 30,
      }
    }
  >
    Només mostra exàmens en hores no lectives i dies festius fins a 3 mesos endevant.     
  </Text>
</ScrollViewMock>
`;

exports[`Testing render from Events component renders view with events and examens 1`] = `
<ScrollViewMock
  style={
    Object {
      "backgroundColor": "#fafafa",
      "flex": 1,
      "paddingTop": 10,
    }
  }
>
  <Text
    style={
      Object {
        "margin": 10,
        "marginBottom": 30,
      }
    }
  >
    Només mostra exàmens en hores no lectives i dies festius fins a 3 mesos endevant.     
  </Text>
</ScrollViewMock>
`;
